pkg/cfn/builder/karpenter.go

```diff
@@ -95,12 +96,8 @@ func (k *KarpenterResourceSet) addResourcesForKarpenter() error {
 
 	roleRef := k.newResource(KarpenterNodeRoleName, &role)
 
-	instanceProfileName := gfnt.MakeFnSubString(fmt.Sprintf("eksctl-%s-%s", KarpenterNodeInstanceProfile, k.clusterSpec.Metadata.Name))
```

Comment:
Removed this so it's not created in two places lest we forget to update it if it ever changes in style.